ocs_ci/ocs/cluster_load.py

```diff
@@ -4,277 +4,335 @@
+            # diff between FIO pod creation iterations
+            iops_diff = (current_iops / previous_iops * 100) - 100
+            low_diff_counter += 1 if -15 < iops_diff < 10 else 0
+            if low_diff_counter > 3:
+                logger.warning(
+                    f"Limit was determined by low IOPS diff between "
+                    f"iterations - {iops_diff:.2f}%"
+                )
+                limit_reached = True
 
-        current_throughput = self.cl_obj.calc_average_throughput()
+            if time.time() > time_before + time_to_wait:
+                logger.warning(
+                    f"Could not determine the cluster IOPS limit within"
+                    f"\nthe given {time_to_wait} seconds timeout. Breaking"
+                )
+                limit_reached = True
 
-        if target_throughput:
-            if current_throughput > target_throughput * 0.8:
-                self.logger.info(
-                    f"Cluster limit has been reached. It is {current_throughput} Mb/s"
+            cluster_used_space = get_percent_used_capacity()
+            if cluster_used_space > 60:
+                logger.warning(
+                    f"Cluster used space is {cluster_used_space}%. Could "
+                    f"not reach the cluster IOPS limit before the "
+                    f"used spaced reached 60%. Breaking"
                 )
                 limit_reached = True
 
-        time_to_wait = 60 * 30
-        time_before = time.time()
-        if not target_throughput:
-            self.logger.info(
-                f"\n=====================================================================\n"
-                f"Determining the cluster throughput limit. Once determined, IOs will be"
-                f"\nreduced to load at {target_percentage * 100}% of the cluster limit"
-                f"\n====================================================================="
+        self.cluster_limit = cluster_limit
+        logger.info(
+            f"\n===================================\nThe cluster IOPS limit "
+            f"is {self.cluster_limit:.2f}\n==================================="
+        )
+        logger.info(
+            f"Deleting all DC FIO pods that have FIO rate parameter of {rate}"
+        )
+        while self.dc_objs:
+            self.decrease_load(wait=False)
+
+        # Creating the first pod of small FIO 'rate' param, to speed up the process.
+        # In the meantime, the load will drop, following the deletion of the
+        # FIO pods with large FIO 'rate' param
+        rate = '15M'
+        self.increase_load(rate=rate)
+        target_iops = self.cluster_limit * self.target_percentage
+        current_iops = self.get_query(query=constants.IOPS_QUERY)
+        logger.info(f"Target IOPS: {target_iops}")
+        logger.info(f"Current IOPS: {current_iops}")
+
+        while current_iops < target_iops * 0.95:
+            logger.info(
```

Comment:
This log message should be probably before first load is started (line 237)

Reply:
correct, will fix